app/models/mixins/retirement_mixin.rb

```diff
@@ -39,6 +39,10 @@ def set_retirement_requester(obj_ids, requester)
     end
   end
 
+  def make_retire_request(_src_ids, requester, _initiated_by)
```

Comment:
last thing - maybe just `make_retire_request(requester, initiated_by: 'user')` as we have id from instance 


and you can also pass initiated_by to self.class.make_retire_request`